modin/pandas/test/test_io.py

```diff
@@ -1036,6 +1036,40 @@ def test_from_csv(make_csv_file, nrows):
     df_equals(modin_df, pandas_df)
 
 
+def dummy_decorator():
+    """A problematic decorator that does not use `functools.wraps`. this introduce unwanted local variables for
+    inspect.currentframe.
+    """
+
+    def wrapper(method):
+        def wrapped_function(self, *args, **kwargs):
+            result = method(self, *args, **kwargs)
+            return result
+
+        return wrapped_function
+
+    return wrapper
+
+
+@pytest.mark.parametrize("nrows", [123, None])
+def test_from_csv_within_decorator_parse_args_correctly(make_csv_file, nrows):
+    make_csv_file()
+
+    @dummy_decorator()
+    def wrapped_read_csv(file, nrows):
+        return pd.read_csv(file, nrows=nrows)
+
+    pandas_df = pandas.read_csv(TEST_CSV_FILENAME, nrows=nrows)
```

Comment:
It is better to to perform tests with the same conditions (if it is possible), for example
```
    @dummy_decorator()
    def wrapped_read_csv(file, nrows, pd):
        return pd.read_csv(file, nrows=nrows)

    pandas_df = wrapped_read_csv(TEST_CSV_FILENAME, nrows=nrows, pd=pandas)
    modin_df = wrapped_read_csv(TEST_CSV_FILENAME, nrows=nrows, pd=pd)
```

Reply:
make sense. changed.